index.html

```diff
@@ -1318,19 +1318,17 @@ <h4 id="h-duration-change-algorithm" resource="#h-duration-change-algorithm"><sp
           <p>Follow these steps when <code><a href="#widl-MediaSource-duration">duration</a></code> needs to change to a <var>new duration</var>.</p>
           <ol>
             <li>If the current value of <code><a href="#widl-MediaSource-duration">duration</a></code> is equal to <var>new duration</var>, then return.</li>
-            <li>Set <var>old duration</var> to the current value of <code><a href="#widl-MediaSource-duration">duration</a></code>.
-            </li><li>Update <code><a href="#widl-MediaSource-duration">duration</a></code> to <var>new duration</var>.</li>
-            <li>If the <var>new duration</var> is less than <var>old duration</var>, then run the
-              <a href="#sourcebuffer-range-removal">range removal</a> algorithm with <var>new duration</var> and
-              <var>old duration</var> as the start and end of the removal range.
-
-              <div class="note"><div class="note-title marker" aria-level="5" role="heading" id="h-note15"><span>Note</span></div><p class="">This preserves audio frames and text cues that start before and end after the <code><a href="#widl-MediaSource-duration">duration</a></code>.</p></div></li>
-            <li>If a user agent is unable to partially render audio frames or text cues that start before and end after the <code><a href="#widl-MediaSource-duration">duration</a></code>, then run the following steps:
+			<li>If <var>new duration</var> is less than the highest starting <a href="#presentation-timestamp">presentation timestamp</a> reported by the <code><a href="#widl-SourceBuffer-buffered">buffered</a></code> attribute across all <a href="#idl-def-SourceBuffer" class="idlType"><code>SourceBuffer</code></a> 
```

Comment:
Is your intent to specify checking **all track buffers** in SourceBuffer objects for this MediaSource.  If so, I agree with that addition.  The language "for all SourceBuffer objects in SourceBuffers" seems equally correct to me though.

I agree that the intent of `remove()` is to remove all frames whose entire presentation interval lies after the new duration, and that the Duration Change condition should be aligned with this.

I'm puzzled by some of your issues though:

1. In your example, is your concern not removing the 0 time coded frame?  It should be removed, since it's timestamp is equal to the remove range start.
2. Are you concerned with using the `buffered` attribute to test the timestamps of buffered data?  I think it may not belong here, and that your language about timestamps for any buffered coded frame across all track buffers would be more correct.  

That might get us to:
> If new duration is less than the highest starting presentation timestamp of any buffered coded frame for all track buffers and SourceBuffer objects in sourceBuffers, then throw an InvalidStateError exception and abort these steps.

This leaves in "starting" to align with this Coded Frame Removal language:

> Remove all media data, from this track buffer, that contain starting timestamps greater than or equal to start and less than the remove end timestamp.

So if we settle whether this is correct or not, I think we can proceed.